### PR TITLE
add delete osm_recipients

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailRegistryService.java [hunk 1]
```diff
@@ -6,18 +6,30 @@
...
+	private final JdbcTemplate jdbcTemplate;
+	
+	private final OsmRecipientsRepository recipientsRepository;
+	
+	public EmailRegistryService(OsmRecipientsRepository recipientsRepository, JdbcTemplate jdbcTemplate) {
+		this.recipientsRepository = recipientsRepository;
+		this.jdbcTemplate = jdbcTemplate;
+	}
+	
+	@Transactional
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailRegistryService.java:28]: here transactional not needed - check unsubscribedRepo.deleteAllByEmail(email);

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailRegistryService.java [hunk 2]
```diff
@@ -6,18 +6,30 @@
...
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 public class EmailRegistryService {
-
-	@Autowired
-	private JdbcTemplate jdbcTemplate;
-
+	
+	private final JdbcTemplate jdbcTemplate;
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailRegistryService.java:19]: Doesn't follow conventions we use everywhere  Autowired so here it's not clear why should we do different

Comment (alisa911) [reply java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailRegistryService.java:19]: @Autowired isn't recommended. On the official page Spring, they recommend to use Constructor-based Dependency Injection. @Autowired has many restrictions in use, hides interface and other. Intelij Idea official highlights @Autowired us unrecommended.

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailRegistryService.java (hunk 1 above)
Comment (alisa911) [reply java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailRegistryService.java:28]: I added it after exception No EntityManager. I read it need for deleteBy custom method

**Pull request merged**